packages/pwa-kit-react-sdk/src/ssr/universal/hooks/index.js

```diff
@@ -42,3 +43,31 @@ export const useServerContext = () => {
 
     return serverContext
 }
+
+/**
+ * Returns the application's origin.
+ *
+ * By default, it will return the ORIGIN under which we are serving the page.
+ *
+ * If `fromXForwardedHeader` is true, it will use the value of `x-forwarded-proto` and `x-forwarded-host` headers in req
+ * to build origin. (it is false by default)
+ *
+ * NOTE: this is a React hook, so it has to be used in a React rendering pipeline.
+ * @returns {string} origin string
+ *
+ */
+export const useOrigin = ({fromXForwardedHeader = false}) => {
+    const {res} = useServerContext()
```

Comment:
is `res` defined on the client side?

Reply:
Nothing, since res here will be undefined and the code block for client will run. We are not using it yet unless it is on server. You can check the deployed version here: https://www.phased-launch-testing.com/

Reason why I have to move this line to beginning of the hook is due to the react hook conditional rendering rules.